bug: transaction with no input/output states was not delegated properly

### What happened?

I was testing some changes to the Noto domain, part of which involved building a transaction that had only "read" and "info" states, and no "input" or "output" states. The transaction flow seemed to get stuck when it should have delegated to the notary. While I saw the first few steps of the transaction flow happen on node 2 (the sender node), I saw no record at all of the transaction on node 1 (the notary node). When I changed the flow to have "inputs" and "outputs" after assembly, it seemed to work.

### What did you expect to happen?

Transactions should always be delegated and submitted properly, regardless of what types of states they contain.

### How can we reproduce it (as minimally and precisely as possible)?

See attached logs. Transaction was `046a84d1-bf4e-4ab2-8fa7-74844f4a3306`.

[paladin1.log](https://github.com/user-attachments/files/18325669/paladin1.log)
[paladin2.log](https://github.com/user-attachments/files/18325670/paladin2.log)
[paladin3.log](https://github.com/user-attachments/files/18325671/paladin3.log)


### Anything else we need to know?

I'm not positive that this is related to read/info states, but that seems to be the only obvious difference in the flows that worked and those that did not.

### OS version

_No response_

The crash:
```
[2025-01-07T19:55:07.336Z] ERROR Error getting state distributions: PD011831: Invalid transaction state for state distribution pid=1 role=pctm-loop-0xad3c780a19ec30d6d2f49b89acc457855c3be4cf
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xffff46f270a8]

goroutine 1058 [running]:
github.com/kaleido-io/paladin/core/internal/privatetxnmgr.(*transactionFlow).applyTransactionAssembledEvent(0x400096f380, {0xffff47693798, 0x40008b9680}, 0x4000a3cdc0)
        /app/core/go/internal/privatetxnmgr/transaction_flow_mutators.go:125 +0x1b8
github.com/kaleido-io/paladin/core/internal/privatetxnmgr.(*transactionFlow).ApplyEvent(0x400096f380, {0xffff47693798, 0x40008b9680}, {0xffff47693b50, 0x4000a3cdc0})
        /app/core/go/internal/privatetxnmgr/transaction_flow_mutators.go:43 +0x294
github.com/kaleido-io/paladin/core/internal/privatetxnmgr.(*Sequencer).handleTransactionEvent(0x4000cb0600, {0xffff47693798, 0x40008b9680}, {0xffff47693b50, 0x4000a3cdc0})
        /app/core/go/internal/privatetxnmgr/sequencer_event_loop.go:101 +0x1b8
github.com/kaleido-io/paladin/core/internal/privatetxnmgr.(*Sequencer).evaluationLoop(0x4000cb0600)
        /app/core/go/internal/privatetxnmgr/sequencer_event_loop.go:48 +0x270
created by github.com/kaleido-io/paladin/core/internal/privatetxnmgr.(*Sequencer).Start in goroutine 1150
        /app/core/go/internal/privatetxnmgr/sequencer.go:325 +0x108
```

--- a/core/go/internal/domainmgr/event_indexer.go
+++ b/core/go/internal/domainmgr/event_indexer.go
@@ -357,7 +357,7 @@ func (d *domain) handleEventBatchForContract(ctx context.Context, dbTX *gorm.DB,
 	}
 
 	// Then any finalizations of those states
-	if len(stateSpends) > 0 || len(stateReads) > 0 || len(stateConfirms) > 0 {
+	if len(stateSpends) > 0 || len(stateReads) > 0 || len(stateConfirms) > 0 || len(stateInfoRecords) > 0 {
 		if err := d.dm.stateStore.WriteStateFinalizations(ctx, dbTX, stateSpends, stateReads, stateConfirms, stateInfoRecords); err != nil {
 			return nil, err
 		}

--- a/core/go/internal/privatetxnmgr/transaction_flow_actions.go
+++ b/core/go/internal/privatetxnmgr/transaction_flow_actions.go
@@ -370,7 +370,8 @@ func (tf *transactionFlow) writeAndLockStates(ctx context.Context) {
 	// we are accessing the transactionFlow's PrivateTransaction object which is only safe to do on the sequencer thread
 	// but we need to make sure that these writes/locks are complete before the assemble requester thread can proceed to assemble
 	// the next transaction
-	if tf.transaction.PostAssembly.OutputStatesPotential != nil && tf.transaction.PostAssembly.OutputStates == nil {
+	if (tf.transaction.PostAssembly.OutputStatesPotential != nil && tf.transaction.PostAssembly.OutputStates == nil) ||
+		(tf.transaction.PostAssembly.InfoStatesPotential != nil && tf.transaction.PostAssembly.InfoStates == nil) {
 		// We need to write the potential states to the domain before we can sign or endorse the transaction
 		// but there is no point in doing that until we are sure that the transaction is going to be coordinated locally
 		// so this is the earliest, and latest, point in the flow that we can do this

